### PR TITLE
doc: update branch prediction docs

### PR DESCRIPTION
**Pull request by Allsimon**

### CODE REVIEW

File: doc/cpu/branch/branch-prediction-fr.md [hunk 1]
```diff
@@ -130,20 +132,20 @@ On part d'un tableau d'entier qui contient des nombres aléatoire compris entre
...
-        for (int value : array) {
-          if (value >= 0) {
-            sum += value;
-          }
-        }
-        return sum;
+int[] array = IntStream.generate(() -> rnd.nextInt() % 1_000)
+    .limit(30_000)
+    .toArray();
+int[] sortedArray = IntStream.of(input).sorted().toArray();
```
Comment (sir4ur0n) [inline doc/cpu/branch/branch-prediction-fr.md]: Shouldn't it be `array` rather than `input`?

File: doc/cpu/branch/branch-prediction-fr.md [hunk 2]
```diff
@@ -130,20 +132,20 @@ On part d'un tableau d'entier qui contient des nombres aléatoire compris entre
...
-
-    long sumPositive(int[] array) {
-        long sum = 0;
-        for (int value : array) {
-          if (value >= 0) {
-            sum += value;
-          }
-        }
-        return sum;
+int[] array = IntStream.generate(() -> rnd.nextInt() % 1_000)
```
Comment (sir4ur0n) [inline doc/cpu/branch/branch-prediction-fr.md]: Consider renaming `array` to `unsortedArray` for consistency with the table below

File: doc/cpu/branch/branch-prediction-fr.md (hunk 1 above)
Comment (Allsimon) [reply doc/cpu/branch/branch-prediction-fr.md]: thanks, done :)

File: doc/cpu/branch/branch-prediction-fr.md (hunk 2 above)
Comment (Allsimon) [reply doc/cpu/branch/branch-prediction-fr.md]: thanks, done :)

**Review by sir4ur0n — APPROVED** [review]

**Pull request merged**